File: src/main/java/model/equipment/Equipment.java
package model.equipment;

import model.support.allEnums.EquipmentType;

public abstract class Equipment {
    protected EquipmentType equipmentType;
    protected double quality;
    protected double price;
    protected double weight;

    public Equipment(EquipmentType equipmentType) {
        this.equipmentType = equipmentType;
    }

    public double getQuality() {
        return quality;
    }

    public double getPrice() {
        return price;
    }

    public double getWeight() {
        return weight;
    }

    public EquipmentType getEquipmentType() {
        return equipmentType;
    }
}

File: src/main/java/model/equipment/Equipment.kt
package model.equipment

import model.support.enums.EquipmentType

abstract class Equipment(val equipmentType: EquipmentType, weight: Double = 0.0, quality: Int = 0, price: Double = 0.0) {
    var weight = weight
        protected set
    var quality = quality
        protected set
    var price = if (price < 0) 0.0 else price
        protected set

    companion object {
        val NONE = object : Equipment(EquipmentType.NONE) {}
    }
}
